nft_ingester/src/program_transformers/bubblegum/db.rs

```diff
@@ -532,70 +532,46 @@ pub async fn upsert_asset_creators<T>(
 where
     T: ConnectionTrait + TransactionTrait,
 {
-    // Vec to hold base creator information.
-    let mut db_creators = Vec::with_capacity(creators.len());
-
-    if creators.is_empty() {
-        // Bubblegum supports empty creator array.  In this case insert an empty Vec
-        // for the creator.
-        db_creators.push(asset_creators::ActiveModel {
+    let db_creators = creators
+        .iter()
+        .enumerate()
+        .map(|(i, creator)| asset_creators::ActiveModel {
             asset_id: Set(id.clone()),
-            position: Set(0),
-            creator: Set(vec![]),
-            share: Set(100),
-            verified: Set(false),
+            position: Set(i as i16),
+            creator: Set(creator.address.to_bytes().to_vec()),
+            share: Set(creator.share as i32),
+            verified: Set(creator.verified),
             slot_updated: Set(Some(slot_updated)),
             seq: Set(Some(seq)),
             ..Default::default()
-        });
-    } else {
-        // Set to prevent duplicates.
-        let mut creators_set = HashSet::new();
-
-        for (i, c) in creators.iter().enumerate() {
-            if creators_set.contains(&c.address) {
-                continue;
-            }
```

Comment:
But I think you are right and we don't need the HashSet here any more as the contracts enforce uniqueness and we don't require it.